CustomCollections/CustomCollections/CustomHashTable.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace CustomCollections
+{
+    class CustomHashTable<TKey, TValue> : IDictionary<TKey, TValue>
+    {
+        private static KeyValuePair<TKey, TValue>[] _hashTable;
+
+        public int Count { get; private set; } = 0;
+
+        public int Capacity => _hashTable.Length;
+
+        public bool IsReadOnly => false;
+
+        public ICollection<TKey> Keys => GetKeys();
+        public ICollection<TValue> Values => GetValues();
+
+        public CustomHashTable() => _hashTable = new KeyValuePair<TKey, TValue>[2];
+
+        public void Clear()
+        {
+            _hashTable = new KeyValuePair<TKey, TValue>[2];
+            Count = 0;
+        }
+
+        public bool Contains(KeyValuePair<TKey, TValue> item)
+        {
+            if (item.Key == null)
+            {
+                throw new ArgumentNullException("key");
+            }
+            var position = GetPosition(item.Key);
+            return EqualityComparer<KeyValuePair<TKey, TValue>>.Default.Equals(_hashTable[position], item);
```

Comment:
You use this `EqualityComparer<KeyValuePair<TKey, TValue>>.Default` all. Can't you store that into a private variable with simple name?